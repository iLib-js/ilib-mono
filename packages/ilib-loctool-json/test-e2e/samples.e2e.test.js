/*
 * samples.e2e.test.js - E2E tests for ilib-loctool-json samples
 *
 * Copyright Â© 2025, Box, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

const path = require("path");
const fs = require("fs");
const { expectFileToMatchSnapshot, LoctoolRunner } = require("ilib-internal");

describe("samples", () => {
    describe("js-json", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "js-json");
        const xliffPath = path.resolve(projectPath, "sample-js-json-extracted.xliff");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            // Clean up all generated files and directories
            const filesToClean = [
                xliffPath, // sample-js-json-extracted.xliff
                path.resolve(projectPath, "sample-js-json-new-de-DE.xliff"),
                path.resolve(projectPath, "sample-js-json-new-ko-KR.xliff"),
                path.resolve(projectPath, "resources")
            ];

            filesToClean.forEach(file => {
                if (fs.existsSync(file)) {
                    if (fs.statSync(file).isDirectory()) {
                        fs.rmSync(file, { recursive: true, force: true });
                    } else {
                        fs.unlinkSync(file);
                    }
                }
            });
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });

    describe("json", () => {
        const projectPath = path.resolve(__dirname, "..", "samples", "json");
        const xliffPath = path.resolve(projectPath, "sample-json-extracted.xliff");

        beforeAll(async () => {
            const loctool = new LoctoolRunner(projectPath);
            await loctool.run("localize");
        });

        afterAll(() => {
            // Clean up all generated files and directories
            const filesToClean = [
                xliffPath, // sample-json-extracted.xliff
                path.resolve(projectPath, "sample-json-new-de-DE.xliff"),
                path.resolve(projectPath, "sample-json-new-ko-KR.xliff"),
                path.resolve(projectPath, "i18n/de-DE.json"),
                path.resolve(projectPath, "i18n/ko-KR.json")
            ];

            filesToClean.forEach(file => {
                if (fs.existsSync(file)) {
                    if (fs.statSync(file).isDirectory()) {
                        fs.rmSync(file, { recursive: true, force: true });
                    } else {
                        fs.unlinkSync(file);
                    }
                }
            });
        });

        it("should produce an extracted XLIFF file", () => {
            expectFileToMatchSnapshot(xliffPath);
        });
    });
});
