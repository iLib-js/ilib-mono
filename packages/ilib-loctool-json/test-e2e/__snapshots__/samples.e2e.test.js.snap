// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samples js-json should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="src/t1.js" source-language="en-US" product-name="sample-js-json">
    <body>
      <trans-unit id="1" resname="Hello" restype="string" datatype="javascript">
        <source>Hello</source>
      </trans-unit>
      <trans-unit id="2" resname="This is a string that has no translation yet" restype="string" datatype="javascript">
        <source>This is a string that has no translation yet</source>
      </trans-unit>
      <trans-unit id="3" resname="Thank you" restype="string" datatype="javascript">
        <source>Thank you</source>
      </trans-unit>
      <trans-unit id="4" resname="id1" restype="string" datatype="javascript">
        <source>This is a test with a unique id</source>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;

exports[`samples json should produce an extracted XLIFF file 1`] = `
"<?xml version="1.0" encoding="utf-8"?>
<xliff version="1.2">
  <file original="i18n/sample.json" source-language="en-US" product-name="sample-json">
    <body>
      <trans-unit id="1" resname="project.whatever.hello" restype="string" datatype="json">
        <source>Hello</source>
        <note>A greeting</note>
      </trans-unit>
      <trans-unit id="2" resname="project.whatever.world" restype="string" datatype="json">
        <source>World</source>
        <note>An object that greetings are directed to</note>
      </trans-unit>
      <trans-unit id="3" resname="project.whatever.translate.me" restype="string" datatype="json">
        <source>Translate me</source>
        <note>This is a string that has no translation yet</note>
      </trans-unit>
      <trans-unit id="4" resname="project.whatever.and.me" restype="string" datatype="json">
        <source>And me</source>
        <note>Another string that has no translation yet</note>
      </trans-unit>
    </body>
  </file>
</xliff>"
`;
